### DISCUSSION TITLE
yolo

### DISCUSSION THREAD

**Discussion by keleduishui** [ultralytics/ultralytics#16416] [Q&A]
你好，如果我的数据集图片大小为2048*1000，那么我将该图片进行缩放和填充操作使其大小为640*640，这样是否意味着我的标签数据也要进行修改？应该如何进行修改呢？

**Comment by UltralyticsAssistant** [comment]
👋 Hello @keleduishui, thank you for reaching out to Ultralytics 🚀! An Ultralytics engineer will assist you soon. In the meantime, here are some resources that might help:

We recommend visiting the [Docs](https://docs.ultralytics.com) for guidance. You'll find useful information about [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, which could have answers to your query.

If your question is regarding data resizing and label adjustment, please ensure you check our [data preparation guidelines](https://docs.ultralytics.com/tutorials/data-preparation/). Adjusting labels is necessary when images are resized and padded, as this might affect the coordinates in your annotations.

If this is about a specific implementation or a 🐛 bug, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us understand the issue better.

Feel free to join our community for real-time discussions on [Discord](https://ultralytics.com/discord) 🎧 or dive into more structured conversations on [Discourse](https://community.ultralytics.com).

## Upgrade

Ensure you've upgraded to the latest `ultralytics` package:

```bash
pip install -U ultralytics
```

## Environments

You can run YOLOv8 in verified environments:

- **Notebooks** with free GPU: 
  - <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a>
  - <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>
  - <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Check this badge to see if all tests are passing. It verifies correct operation across all modes and tasks every 24 hours on multiple platforms.

**Comment by glenn-jocher** [comment]
你好，是的，缩放和填充图片后，标签数据也需要相应调整。可以使用 YOLO 的数据预处理功能来自动处理这些变化。

**Reply by keleduishui** [reply]
你好，我对yolo不太了解，请问yolo的数据预处理功能是什么意思

**Reply by glenn-jocher** [reply]
@keleduishui yOLO的数据预处理功能包括自动调整图像大小、归一化像素值等，以便更好地进行模型训练和推理。

**Reply by keleduishui** [reply]
yolo本身会自动调整输入图像的尺寸吗？是不是得我事先对数据进行手动缩放的预处理

**Reply by glenn-jocher** [reply]
YOLO会自动调整输入图像的尺寸，无需手动预处理。

**Reply by keleduishui** [reply]
嗨你好，我应该如何冻结yolov8的backbon部分，以及我在ultralytics的哪里可以了解到冻结yolo的知识

**Reply by glenn-jocher** [reply]
你好，可以通过设置`freeze`参数来冻结YOLOv8的backbone部分。详细信息可以在Ultralytics的文档中找到。

**Reply by keleduishui** [reply]
你好，我需要对yolo进行迁移学习，在预训练时yolo检测的类别为7，但是经过迁移学习后yolo的检测类别为5，请问我需要更换全连接层吗？如果需要更换请问应该使用什么代码来更换全连接层？

**Reply by glenn-jocher** [reply]
@keleduishui 你好，YOLO在迁移学习时会自动调整类别数量，无需手动更换全连接层。只需在数据配置文件中更新类别数即可。
